tests/unit/httpd_test.cc

```diff
@@ -954,6 +954,74 @@ SEASTAR_TEST_CASE(test_client_response_parse_error) {
     });
 }
 
+SEASTAR_TEST_CASE(test_client_request_send_timeout_cached_conn) {
+    return seastar::async([] {
+        loopback_connection_factory lcf(1);
+        auto ss = lcf.get_server_socket();
+        promise<> server_started;
+        future<> server = ss.accept().then([&] (accept_result ar) {
+            server_started.set_value();
+            return seastar::async([sk = std::move(ar.connection)] () mutable {
+                input_stream<char> in = sk.input();
+                read_simple_http_request(in);
+                sleep(std::chrono::seconds(1)).get();
```

Comment:
Nitpick: Although "1 second should be enough for everyone" we always regret this later... Can this be some sort of condition variable or something, to tell this server to just hang forever until released and not for 1 second?

Reply:
Yes, good point. Maybe in this place we can do it.